wagtail/images/tests/test_models.py

```diff
@@ -520,6 +521,12 @@ def setUp(self):
     def test_get_rendition_model(self):
         self.assertIs(Image.get_rendition_model(), Rendition)
 
+    def test_stock_rendition_contains_unique_together_constraint(self):
+        self.assertEqual([], Rendition.check())
+
+    def test_custom_rendition_may_have_unique_constraint(self):
+        self.assertEqual([], CustomRendition.check())
```

Comment:
Unfortunately these only test the "happy path" where either `unique_together` or `CustomRendition` is defined, but not when none of them are defined and the warning is triggered. I think we can follow the approach in #11112 where we dynamically modify the model's `._meta`, call `.check()`, and revert the modifications.